aten/src/ATen/gen.py

```diff
@@ -44,16 +44,27 @@ def will_write(self, filename):
                             "before running the generators")
         self.filenames.add(filename)
 
+    def _write_if_changed(self, filename, contents):
```

Comment:
Is it OK in Makefiles to list something as an output but then not touch it? cmake Makefile target will still use this code in the caffe2 build, for instance. Won't the Makefile continually  try to re-run gen.py on every make because the outputs are older than gen?

Reply:
If this is an actual problem, we can workaround by adding it as an option to the gen script and then only enabling it when ninja is being used.

Reply:
Nope, Makefile knows how to do this correctly.

```
MacBook-Pro-97:make-restat ezyang$ cat Makefile
out: in
	if ! [ -e out ]; then touch out; fi

out2: out
	cp out out2
MacBook-Pro-97:make-restat ezyang$ touch in
MacBook-Pro-97:make-restat ezyang$ make out2
if ! [ -e out ]; then touch out; fi
cp out out2
MacBook-Pro-97:make-restat ezyang$ make out2
make: `out2' is up to date.
MacBook-Pro-97:make-restat ezyang$ touch in
MacBook-Pro-97:make-restat ezyang$ make out2
if ! [ -e out ]; then touch out; fi
```

In fact, the reason why Chrome build system did the no-touch strategy in the first place was because they were originally using CMake (and thus make) for builds.